monkey/infection_monkey/master/plugin_registry.py

```diff
@@ -29,9 +31,13 @@ def get_plugin(self, plugin_name: str, plugin_type: PluginType) -> Any:
         try:
             plugin = self._registry[plugin_type][plugin_name]
         except KeyError:
-            raise UnknownPluginError(
-                f"Unknown plugin '{plugin_name}' of type '{plugin_type.value}'"
-            )
+            response = self._island_api_client.get_agent_plugin(plugin_type, plugin_name)
```

Comment:
I think we want to load it as well, because the same plugin will be fetched more than once. Not sure it that wouldn't cause a race condition though

Reply:
Leaving it as is based on the discussion on Slack.